frontend/tests/features/wheresMyGene.test.ts

```diff
@@ -1143,13 +1207,21 @@ async function getButtonAndClick(page: Page, selector: string) {
   );
 }
 
-async function getNthButtonAndClick(page: Page, selector: string, n: number) {
+async function getCellTypeFmgButtonAndClick(page: Page, cellType: string) {
+  const selector = getTestID(MARKER_GENE_BUTTON_TEST_ID);
+  await waitForElement(page, getTestID(CELL_TYPE_LABELS_ID));
   await tryUntil(
     async () => {
       const buttons = await page.$$(selector);
-      await expect(buttons.length).toBeGreaterThan(n);
-      const button = buttons[n];
-      await button?.click();
+      for (let i = 0; i < buttons.length; i++) {
+        const button = buttons[i];
+        const parentCellTypeSpan = await button.$("xpath=../../..");
+        const text = await parentCellTypeSpan?.innerText();
+        if (text === cellType) {
+          await button.click();
+          return;
+        }
+      }
```

Comment:
Playwright recommends using `.filter()` to find the sub selector you want, but I'm not sure which Playwright version introduced this feature, so our version might not have it yet:

https://playwright.dev/docs/locators#filter-by-another-locator

If we can't use `.filter()` then the old way of [intermediate selector](https://playwright.dev/docs/other-locators#intermediate-matches) `*` should work too. Like so:

```ts
`*css=button >> text=${cellType}`
```

This should select the button that has the text === ${cellType}

Reply:
thanks! i'll update :)